nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -1365,44 +1447,105 @@ impl<'a> Planner<'a> {
                             // The inventory reports this zone but there was an
                             // error reconciling it (most likely an error
                             // starting the zone).
-                            Some(ZoneCurrentlyUpdating {
+                            Some(ZonePropagationIncomplete {
                                 zone_id: zone.id,
                                 zone_kind: zone.kind(),
-                                reason: UpdatingReason::ReconciliationError {
-                                    bp_image_source: &zone.image_source,
-                                    inv_image_source,
-                                    message,
-                                },
+                                reason:
+                                    ZonePropagationStatus::ReconciliationError {
+                                        bp_image_source: &zone.image_source,
+                                        inv_image_source,
+                                        message,
+                                    },
                             })
                         }
                         None => {
                             // The blueprint has a zone that inventory does not have.
-                            Some(ZoneCurrentlyUpdating {
+                            Some(ZonePropagationIncomplete {
                                 zone_id: zone.id,
                                 zone_kind: zone.kind(),
-                                reason: UpdatingReason::MissingInInventory {
-                                    bp_image_source: &zone.image_source,
-                                },
+                                reason:
+                                    ZonePropagationStatus::MissingInInventory {
+                                        bp_image_source: &zone.image_source,
+                                    },
                             })
                         }
                     }
                 })
                 .collect::<Vec<_>>();
+            updating.append(&mut zones_currently_updating);
+        }
+        updating
+    }
 
-            if !zones_currently_updating.is_empty() {
-                info!(
-                    self.log, "some zones not yet up-to-date";
-                    "sled_id" => %sled_id,
-                    "zones_currently_updating" => ?zones_currently_updating,
-                );
-                return Ok(report);
+    /// Update at most one existing zone to use a new image source.
+    fn do_plan_zone_updates(
+        &mut self,
+        mgs_updates: &PlanningMgsUpdatesStepReport,
+    ) -> Result<PlanningZoneUpdatesStepReport, Error> {
+        let mut report = PlanningZoneUpdatesStepReport::new();
+
+        let zones_currently_updating =
+            self.get_zones_not_yet_propagated_to_inventory();
+        if !zones_currently_updating.is_empty() {
+            info!(
```

Comment:
It's a little surprising there's no update to `report` for this.  But maybe that happens elsewhere?

Reply:
I think this is a problem on main too. Filed https://github.com/oxidecomputer/omicron/issues/9047.